includes/collection/class-inbox.php

```diff
@@ -358,4 +372,45 @@ public static function get_by_guid_and_recipient( $guid, $user_id ) {
 
 		return $post;
 	}
+
+	/**
+	 * Deduplicate inbox items with the same GUID.
+	 *
+	 * If multiple inbox items exist with the same GUID (due to race conditions),
+	 * this merges all recipients into the first post and deletes duplicates.
+	 *
+	 * @param string $guid The activity GUID.
+	 *
+	 * @return \WP_Post|false The primary inbox post, or false if no posts found.
+	 */
+	public static function deduplicate( $guid ) {
+		$duplicates = \get_posts(
+			array(
+				'post_type'      => self::POST_TYPE,
+				'guid'           => $guid,
+				'posts_per_page' => -1,
+				'post_status'    => 'any',
+			)
+		);
+
+		if ( empty( $duplicates ) ) {
+			return false;
+		}
+
+		// If only one post, return it.
+		if ( count( $duplicates ) === 1 ) {
+			return $duplicates[0];
+		}
+
+		// Keep the first post, merge recipients from others and delete.
+		$primary = $duplicates[0];
+
+		foreach ( array_slice( $duplicates, 1 ) as $duplicate ) {
```

Comment:
```suggestion
		}
		
		// Keep the first post, all others are duplicates.
		$primary = array_shift( $duplicates );
		
		foreach ( $duplicates as $duplicate ) {
```
If `$duplicates` is empty it won't iterate over the array